llvm/lib/Target/X86/GISel/X86LegalizerInfo.cpp

```diff
@@ -484,6 +498,19 @@ X86LegalizerInfo::X86LegalizerInfo(const X86Subtarget &STI,
       .widenScalarToNextPow2(1);
 
   // vector ops
+  getActionDefinitionsBuilder(G_BUILD_VECTOR)
```

Comment:
AArch64 legalizes G_BUILD_VECTOR and then tries all kinds of tricks in the selector:
https://github.com/llvm/llvm-project/blob/ac7c482ca581cdab11558b5a37ae5095a1638fd2/llvm/lib/Target/AArch64/GISel/AArch64LegalizerInfo.cpp#L874

Reply:
Yes. To be honest this is unclear. Lowering `G_BUILD_VECTOR` earlier should allow to combine loads from constant pool. That is too late for instruction selection. However there may be some scenarios when we want to keep `G_BUILD_VECTOR` to combine into something more optimal. I think we should address to this issue later, when we have more complete picture how we want to optimize various shuffles and insertions. Current approach allows to reuse `G_LOAD` C++ implementation for `G_BUILD_VECTOR` instead of writing its own.